src/views/DragAndDropController.ts

```diff
@@ -37,4 +60,154 @@ export class DragAndDropController implements TreeDragAndDropController<Explorer
+     * @param node the dragged node.
+     */
+    private isDraggableNode(node: DataNode | undefined): boolean {
+        if (!node?.uri) {
+            return false;
+        }
+        if (node instanceof WorkspaceNode || node instanceof ProjectNode
+                || node instanceof PackageRootNode || node instanceof ContainerNode
+                || node instanceof BaseSymbolNode) {
+            return false;
+        }
+
+       return this.isUnderSourceRoot(node);
+    }
+
+    /**
+     * Check whether the node is under source root.
+     *
+     * Note: There is one exception: The primary type directly under an unmanaged folder project,
+     * in that case, `true` is returned.
+     * @param node DataNode
+     */
+    private isUnderSourceRoot(node: DataNode): boolean {
+        let parent = node.getParent();
+        while (parent) {
+            if (parent instanceof ContainerNode) {
+                return false;
+            }
+
+            if (parent instanceof PackageRootNode) {
+                return parent.isSourceRoot();
+            }
+            parent = parent.getParent();
+        }
+        return true;
+    }
+
+    /**
+     * Check whether the node is able to be dropped.
+     */
+    private isDroppableNode(node: ExplorerNode | undefined): boolean {
+        // drop to root is not supported yet
+        if (!node) {
+            return false;
+        }
+
+        if (node instanceof DataNode && !node.uri) {
+            return false;
+        }
+
+        if (node instanceof WorkspaceNode || node instanceof ProjectNode
+                || node instanceof BaseSymbolNode) {
+            return false;
+        }
+
+        let parent: ExplorerNode | undefined = node;
+        while (parent) {
+            if (parent instanceof ProjectNode) {
+                return false;
+            } else if (parent instanceof PackageRootNode) {
+                return parent.isSourceRoot();
+            } else if (parent instanceof ContainerNode) {
+                if (parent.getContainerType() === ContainerType.ReferencedLibrary) {
+                    return (parent.getParent() as ProjectNode).isUnmanagedFolder();
+                }
+                return false;
+            }
+            parent = parent.getParent();
+        }
+        return false;
+    }
+
+    /**
+     * Trigger a workspace edit that move the source node into the target node.
+     */
+    private async move(source: DataNode, target: DataNode): Promise<void> {
+        const sourceUri = Uri.parse(source.uri!);
+        const targetUri = Uri.parse(target.uri!);
+        if (sourceUri === targetUri) {
+            return;
+        }
+
+        const newPath = path.join(targetUri.fsPath, path.basename(sourceUri.fsPath));
+        const choice = await window.showInformationMessage(
+            `Are you sure you want to move '${path.basename(sourceUri.fsPath)}' into '${path.basename(targetUri.fsPath)}'?`,
```

Comment:
When moving a class to another package, it seems not to change the package declaration, the same as dragging in the file explorer.

The Java language server supports move refactoring which will rename the package declaration and reference as well, is it possible we trigger this refactoring for class dragging to package here?

Reply:
That's a regression bug in 1.7.0: https://github.com/redhat-developer/vscode-java/issues/2503

you can build a private build with the latest code or wait for the next pre-release. 😄